pkg/kclient/deployments.go

```diff
@@ -185,18 +190,40 @@ See below for a list of failed events that occured more than %d times during dep
 	}
 }
 
-// CreateDeployment creates a deployment based on the given deployment spec
+func resourceAsJson(resource interface{}) string {
+	data, _ := json.MarshalIndent(resource, " ", " ")
+	return string(data)
+}
+
+// ApplyDeployment updates a deployment based on the given deployment spec
 func (c *Client) CreateDeployment(deploy appsv1.Deployment) (*appsv1.Deployment, error) {
-	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Create(&deploy)
+	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Create(context.TODO(), &deploy, metav1.CreateOptions{FieldManager: "odo"})
```

Comment:
Would it add value to store `metav1.CreateOptions{FieldManager: "odo"}` in a variable? I'm asking because:

```sh
$ git diff main pkg/ | grep "metav1.CreateOptions{FieldManager: \"odo\"}" | wc -l
19
```

Reply:
don't think it makes sense to put the whole `metav1.CreateOptions` into a variable. It would make the code less readable, as you would have to jump to different code sections to see create options. 

I could put `odo` string for `FieldManager` into a separate value.

But for some reason, the way it is written right now felt right to me.